food-truck-frontend/src/components/route/RouteList.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  Button,
+  CircularProgress,
+  Container,
+  Typography,
+} from "@material-ui/core";
+import React from "react";
+import api from "../../util/api";
+import RouteListRow from "../RouteListRows";
+import CreateRouteDialog from "./CreateRouteDialog";
+
+interface RouteListTruck {
+  id: string;
+  userId: string;
+  name: string;
+  schedule: string;
+}
+interface RouteListRouteInfo {
+  routeId: number;
+  routeName: string;
+  active: string;
+}
+
+const TruckObject = {
+  id: "",
+  userId: "",
+  name: "",
+  schedule: "",
+};
+
+const RouteObject = [
+  {
+    routeId: 0,
+    routeName: "",
+    active: "",
+  },
+];
+
+type RouteState = {
+  truck: RouteListTruck | undefined;
+  routeData: RouteListRouteInfo[];
+
+  errorMsg: string | undefined;
+  createRoute: boolean;
+};
+
+type RouteProps = {
+  truckId: string;
+};
+
+class RouteList extends React.Component<RouteProps, RouteState> {
+  constructor(props: RouteProps) {
+    super(props);
+    this.state = {
+      truck: undefined,
+      routeData: RouteObject,
+      errorMsg: undefined,
+      createRoute: false,
+    };
+
+    this.newRoute = this.newRoute.bind(this);
+    this.createSuccess = this.createSuccess.bind(this);
+    this.createFailure = this.createFailure.bind(this);
```

Comment:
Just as a tip, if you declare the functions like this: `newRoute = () => { ... }` then you don't have to do the "bind this" thing.
This way is perfectly fine, though.

Reply:
I know I've always shied away from defining `() =>` functions (arrow functions, I think they're called?) inline because then each render makes a new object, whereas bind creates only one. I'll need to look into whether defining them as `const myFunkyFunc = () => { ... }` is even better, though.